pkg/ingestor/ingestor.go

```diff
@@ -69,6 +69,50 @@ func Ingest(ctx context.Context, d *processor.Document, graphqlEndpoint string,
 	return nil
 }
 
+func BulkIngest(ctx context.Context, docs []*processor.Document, graphqlEndpoint string, csubClient csub_client.Client) error {
+	logger := logging.FromContext(ctx)
+	// Get pipeline of components
+	processorFunc := GetProcessor(ctx)
+	ingestorFunc := GetIngestor(ctx)
+	collectSubEmitFunc := GetCollectSubEmit(ctx, csubClient)
+	assemblerFunc := GetAssembler(ctx, graphqlEndpoint)
+
+	start := time.Now()
+
+	var predicates = make([]assembler.IngestPredicates, 1)
+	var idstrings []*parser_common.IdentifierStrings
+	for _, d := range docs {
+		docTree, err := processorFunc(d)
+		if err != nil {
+			return fmt.Errorf("unable to process doc: %v, format: %v, document: %v", err, d.Format, d.Type)
+		}
+
+		preds, idstrs, err := ingestorFunc(docTree)
+		if err != nil {
+			return fmt.Errorf("unable to ingest doc tree: %v", err)
+		}
+		for i := range preds {
+			predicates[0].CertifyVuln = append(predicates[0].CertifyVuln, preds[i].CertifyVuln...)
+			predicates[0].VulnEqual = append(predicates[0].VulnEqual, preds[i].VulnEqual...)
```

Comment:
Is the idea here that we want to merge the predicates because each doc will only have 1 certify/vuln?

The assemblerFunc we are using is already called "bulk" because it does the bulk ingest per "predicates" Maybe we should call this "MergedIngest"?

Also, we should merge all the predicates here, or name this MergedVulnIngest or similar.

Reply:
> Is the idea here that we want to merge the predicates because each doc will only have 1 certify/vuln?

Each `processor.Document` could potentially generate multiple `assembler.CertifyVulnIngest` because the attestation's statement has an array of subjects.
This is the reason for having the `for` over `preds`.

> The assemblerFunc we are using is already called "bulk" because it does the bulk ingest per "predicates" Maybe we should call this "MergedIngest"?

Yep, sure, makes sense.

> Also, we should merge all the predicates here, or name this MergedVulnIngest or similar.

Fine for merging all predicates here. I didn't do because I wanted to have a feedback but I'm going to add the missing predicates.